libs/gltfio/include/gltfio/ResourceLoader.h

```diff
@@ -55,16 +59,34 @@ struct ResourceConfiguration {
  */
 class ResourceLoader {
 public:
+    using BufferDescriptor = filament::backend::BufferDescriptor;
+
     ResourceLoader(const ResourceConfiguration& config);
     ~ResourceLoader();
+
+    /**
+     * Loads resources for the given asset from the filesystem or data cache and "finalizes" the
+     * asset by transforming the vertex data format if necessary, decoding image files, supplying
+     * tangent data, etc.
+     */
     bool loadResources(FilamentAsset* asset);
+
+    /**
+     * Adds raw resource data into a cache for platforms that do not have filesystem or network
+     * access.
+     */
+    void addResourceData(std::string url, BufferDescriptor&& buffer) {
```

Comment:
Since you marked `mResourceCache` as `mutable`, should this be marked as `const`?

Reply:
Actually it should not be mutable, will fix, thanks for noticing this!